pkg/agent/storage/keyvalue_store_test.go

```diff
@@ -0,0 +1,140 @@
+package storage
+
+import (
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getStores(t *testing.T) []types.KVStore {
+	logger := log.NewNopLogger()
+	db := setupDB(t)
+	bboltStore, err := NewBBoltKeyValueStore(logger, db, "test_bucket")
+	require.NoError(t, err)
+
+	stores := []types.KVStore{
+		NewInMemoryKeyValueStore(logger),
+		bboltStore,
+	}
+	return stores
+}
+
+func Test_GetSet(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		sets        map[string]string
+		gets        map[string]string
+		expectedErr bool
+	}{
+		{
+			name: "empty",
+			sets: map[string]string{},
+			gets: map[string]string{},
+		},
+		{
+			name:        "blank",
+			sets:        map[string]string{"": ""},
+			gets:        map[string]string{"": ""},
+			expectedErr: true,
+		},
+		{
+			name: "multiple",
+			sets: map[string]string{"key1": "value1", "key2": "value2", "key3": "value3"},
+			gets: map[string]string{"key1": "value1", "key2": "value2", "key3": "value3"},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			for _, s := range getStores(t) {
+				for k, v := range tt.sets {
+					err := s.Set([]byte(k), []byte(v))
+					if tt.expectedErr {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
```

Comment:
for some reason I can't stand `else`s =)
```suggestion
					if tt.expectedErr {
						require.Error(t, err)
						continue
					}
					require.NoError(t, err)
```